src/compiler/diagnosticMessages.json

```diff
@@ -1687,6 +1691,10 @@
         "category": "Error",
         "code": 2528
     },
+    "Duplicate identifier '{0}'. Compiler reserves name '{1}' in top level scope of a module containing async functions.": {

```

Comment:
Also "when using async functions" is too broad. It only enforces this check if you use async functions in the current module, hence the wording: "in the top level scope of a module containing async functions."